sycl/plugins/xrt/pi_xrt.cpp

```diff
@@ -675,9 +675,10 @@ void assertion(bool Condition, const char *Message) {
 
 /// Check that we always use the same thread to call this function
 void assert_single_thread() {
-  static std::thread::id saved_id = std::this_thread::get_id();
-  assert(saved_id == std::this_thread::get_id() &&
-         "there is no multithread support for now");
+  [[maybe_unused]] static std::atomic<std::thread::id> saved_id =
+      std::this_thread::get_id();
+  // assert(saved_id == std::this_thread::get_id() &&
```

Comment:
I should probably add a mutex around every calls to the xrt_pi instead of removing this assert.

Reply:
Good idea if you think this is the case. But do you think other PI have the problem? CUDA? HIP? OpenCL? I guess if they do not use mutex it is probably because there is no race condition?